### PR TITLE
[s] Adds a litmus test for servers running open source code

### PR DESCRIPTION
**Pull request by Cyberboss**
@MrStonedOne

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm [hunk 1]
```diff
@@ -265,6 +265,11 @@
 	else
 		..()
 
+/mob/verb/osscheck()
+	set name = "osscheck"
+	set hidden = TRUE
+	alert(usr, "You are running code from an OSS codebase")
+
```
Comment (coiax) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: But in farm animals?

Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: ¯\_(ツ)_/¯

Comment (duncathan) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: nice and hidden so it's unlikely to be removed

Comment (coiax) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: Maybe at least put a comment there so people know why it's there.

**Comment by RandomMarine** [comment]
What exactly is the purpose of this?

**Comment by Supermichael777** [comment]
so as a player with no special privileges you can find this out

**Comment by Anonmare** [comment]
@RandomMarine I assume to ensure there's a quick and easy check to see if another server is using our code without it being obvious to remove.

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm (hunk 1 above)
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: this PR gives it away though.
you can also use the verb and search DM for the string it outputs.

tl;dr bad meme.

Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: The idea is that, by the time we actually need to use it, the vast majority of people will have forgotten about it.

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: or people remove it immediately, now, when it's most obvious and vulnerable.

**Comment by xxalpha** [comment]
Why not obfuscate this, why put it there in plaintext and explain it in the pr

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: Add a header to every file checking for this.

**Comment by coiax** [comment]
I mean, I'm not sure if this is needed. If people lift our codebase, unless they change everything, there will be enough distinctive strings, patterns, behaviour.

**Comment by xxalpha** [comment]
That is also true.

**Comment by AnturK** [comment]
Is this a meme PR ? Because i'm not sure you're serious or i should close it.

**Comment by Cyberboss** [comment]
Does my name say @iamgoofball?

On May 4, 2017 3:26 PM, "AnturK" <notifications@github.com> wrote:

Is this a meme PR ? Because i'm not sure you're serious or i should close
it.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/tgstation/tgstation/pull/26900#issuecomment-299284870>,
or mute the thread

**Comment by RemieRichards** [comment]
yes.

**Comment by AnturK** [comment]
If you can get maintainer insane enough to agree to this feel free to reopen.

**Comment by xxalpha** [comment]
itsgoof

**Comment by Cyberboss** [comment]
Anyone can F12

**Pull request closed without merging**